### PR TITLE
Tic-Tac-Toe Game Milestone 3

### PR DESCRIPTION
**Pull request by Abidoyinsola1**
The following changes are made on this project:

- Created a player object
- Created the game board
- Added game logic

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Nice work on this. 👍

However, let's fix this:

- [x] Please use attr_reader/attr_writer only use attr_accessor when absolutely necessary, be aware that by using it you allow writing that property from outside the class. That variable can be set to anything and pose a risk. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [x] Also the scenario as shown in the screenshot is expected to be a winning position but yet the game still asks for the next player input. Kindly fix it. 👇[Screenshot]
ice_screenshot_20200804-184957

- [x] In addition, when all board positions haven't been filled the game ends abruptly and throws an error. Kindly fix this. 👇[Screenshot]
ice_screenshot_20200804-185521

- [x] Also kindly remove all puts/print expression in `lib` folder. Puts/print only belong in the `bin/main.rb` file.

- [x] Consider outputting a message to the console when there is a draw in the game.


Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: lib/game_board.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+class GameBoard
+  attr_accessor :board, :player1, :player2
```
Comment (mcrd25) [inline lib/game_board.rb]: is `attr_accessor` necessary? or would `attr_reader` or `attr_writer` be more applicable?
See [screenshot(https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) of requirement as previously shown by @mikenath223

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Player
+  attr_accessor :player_name, :symbol, :position
```
Comment (mcrd25) [inline lib/player.rb]: is `attr_accessor` necessary? or would `attr_reader` or `attr_writer` be more applicable?
See [screenshot(https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) of requirement as previously shown by @mikenath223

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required :mag: 

Hi Team! :wave: 

Good job fixing most of the changes requested from the previous review!
However, you are still using `attr_accessor` where `attr_reader` or `attr_writer` would be more applicable 
Please address this!

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

File: lib/game_board.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+class GameBoard
+  attr_accessor :player1, :player2
```
Comment (mosesogwo) [inline lib/game_board.rb]: Please use attr_reader for these attributes. They are not set anywhere after the GameBoard is initialized.
Same goes for the attributes in your Player class.

File: lib/player.rb (hunk 2 above)
Comment (mosesogwo) [inline lib/player.rb]: Please use attr_reader for these attributes. They are not set anywhere after the Player is initialized.

**Review by mosesogwo — APPROVED** [review]
**Hello! :wave: Please see my comment below, effect the change and you can move on.**
Milestone is approved:+1:.

### Status: Approved :heavy_check_mark:

**Pull request merged**